From 744581f19b32f4cbcfd7b839a2a1d9fbe653b250 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 18 Jan 2021 22:49:35 +0200
Subject: [PATCH] [SW-20810] arm64: use strict-align flag

Add strict-align flag to GCC compilation flags. This ensures the GCC is
aware that the CPU is unable to handle access to unaligned data. When
the parser requires to perform unaligned read/write, it will use 8bit
read/write in order to ensure aligned access.

Change-Id: I6276b1b400fde1ce98f813d2546a7295a54c5720
Signed-off-by: Guy Nisan <gnisan@habana.ai>
(cherry picked from commit 350d43a8f26c963c267c80c3ba686bbf61f331d1)
---

diff --git a/gaudi/Makefile b/gaudi/Makefile
index b06b553..699aa0f 100644
--- a/gaudi/Makefile
+++ b/gaudi/Makefile
@@ -6,4 +6,6 @@
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/hbm.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/reset.c)
 
+CFLAGS += -mstrict-align
+
 image_validation: image_validation_common
diff --git a/goya/Makefile b/goya/Makefile
index 1468b5a..0bbb583 100644
--- a/goya/Makefile
+++ b/goya/Makefile
@@ -1,4 +1,6 @@
 FW_SIZE := 0x8000
 PPBOOT_MAX_SIZE := 6144
 
+CFLAGS += -mstrict-align
+
 image_validation: image_validation_common
diff --git a/goya2/Makefile b/goya2/Makefile
index 8f3266a..2158915 100644
--- a/goya2/Makefile
+++ b/goya2/Makefile
@@ -2,4 +2,6 @@
 PPBOOT_MAX_SIZE := 18432
 PREBOOT_SRCS_C += $(shell ls preboot/thermal_v2.c)
 
+CFLAGS += -mstrict-align
+
 image_validation: image_validation_common
